sound/soc/codecs/tas2552.c

```diff
@@ -581,8 +581,8 @@ static int tas2552_component_probe(struct snd_soc_component *component)
 
 	gpiod_set_value(tas2552->enable_gpio, 1);
 
-	ret = pm_runtime_get_sync(component->dev);
-	if (ret < 0) {
+	ret = pm_runtime_resume_and_get(component->dev);
+	if (ret < 0 && ret != -EACCES) {
```

Comment:
Same comment as for McASP: if we get `-EACCES` it means that the device is _not_ enabled...

Reply:
@plbossart, right, I did interpreted it wrong, you are correct. If the pm_runtime_enable() is not called on the device then yes, -EACCES could be returned, but it is called in probe time, so for drivers itself this will never happen, unless the pm_runtime_enable() is conditional in some way, but I don't see that in the tas2552.c nor in davinci-mcasp.c for example.

In framework level this make sense as there might be drivers managed by the framework which does not use runtime_pm.

Reply:
@ujfalusi I added -EACCES by default in all cases, because it's hard to figure out if it's missing by mistake or intentionally (because it cannot happen by design).

However I did confuse pm_runtime_allow/forbid (which is accessible to the user via the 'control' sysfs file) and pm_runtime_enable/disable, which can only be done by drivers. You're absolutely right that -EACCES can only be returned if pm_runtime_enable is missing somehow.

So yes I agree that we should more carefully re-inspect all the -EACCES additional checks when there were missing. I will assume that if it was present before then we leave it as is. 

I will revisit all this more carefully, good feedback